packages/edit-site/src/components/sidebar-navigation-screen-global-styles/index.js

```diff
@@ -41,7 +41,13 @@ export function SidebarNavigationItemGlobalStyles( props ) {
 			/>
 		);
 	}
-	return <SidebarNavigationItem { ...props } />;
+	return (
```

Comment:
Thanks for getting this PR up! I'll test further later today.

I'm wondering if we need the `hasGlobalStyleVariations` condition at all. 🤔 

E.g., 

```jsx
export function SidebarNavigationItemGlobalStyles( props ) {
	const { name } = useLocation();
	return (
		<SidebarNavigationItem
			{ ...props }
			to="/styles"
			uid="global-styles-navigation-item"
			aria-current={ name === 'styles' }
		/>
	);
}
```

I'll investigate later. 

Also, not related to this PR, but there's a persistent JS error for sidebar items on mobile: https://github.com/WordPress/gutenberg/pull/67467#issuecomment-2513472204

Just mentioning in case you see it and are wondering 👍🏻

Reply:
I thought about removing the condition too. But after looking at the original code, I saw that the condition was there and the uid was missing in the default return. To keep the code consistent and working as expected, I decided to leave it as is.

Reply:
Ah, thanks for the context. That's a good point. Anyway, I'll test as soon as I can today and we'll aim to get the fix in ASAP.

Reply:
I still think a lot of this code is superfluous and doubles up on things.

For example, most of the props could be passed to `SidebarNavigationItemGlobalStyles` in [MainSidebarNavigationContent](https://github.com/WordPress/gutenberg/blob/40b865333994a9cc1f239d0f4cfd817be138d4b4/packages/edit-site/src/components/sidebar-navigation-screen-main/index.js#L30-L30).

```jsx
			<SidebarNavigationItemGlobalStyles
				to="/styles"
				uid="global-styles-navigation-item"
				icon={ styles }
			>
				{ __( 'Styles' ) }
			</SidebarNavigationItemGlobalStyles>
```

With the same effect.

Removing the condition doesn't have any side effects as far as I can see, but I agree that we can take the cautious route to fix the bug. 

I can get a follow up to clean up the code and we can get folks who have commented on [recent changes](https://github.com/WordPress/gutenberg/commit/c9a5cab0e2e9063054ed4f299bde734a242e3794#diff-13a4b1af1602640c2d5f796fc64e29f6756168a5299458db6b09f3584cf34024R38) to routing here to chime in.

cc @youknowriad and @t-hamano for visibility